app/gui2/src/components/GraphEditor.vue

```diff
@@ -44,28 +44,37 @@ useWindowEvent('keypress', (e) => {
   if (pos == null) return
 
   switch (e.key) {
-    case 'n':
+    case 'n': {
       const n = graphStore.createNode(pos)
       if (n == null) return
       graphStore.setNodeContent(n, 'hello "world"! 123 + x')
       break
+    }
   }
 })
 
 function updateNodeContent(id: ExprId, range: ContentRange, content: string) {
   graphStore.replaceNodeSubexpression(id, range, content)
 }
+
+function moveNode(id: ExprId, delta: Vec2) {
+  const node = graphStore.nodes.get(id)
+  if (node == null) return
+  const newPosition = node.position.addScaled(delta, 1 / navigator.scale)
+  graphStore.setNodePosition(id, newPosition)
+}
 </script>
 
 <template>
   <div ref="viewportNode" class="viewport" v-on="navigator.events" @mousemove="updateMousePos">
     <svg :viewBox="navigator.viewBox">
       <GraphEdge
-        v-for="edge in graphStore.edges"
+        v-for="(edge, index) in graphStore.edges"
+        :key="index"
```

Comment:
should we be explicitly specifying `:key="index"` for every `v-for`? (not an issue, just a question)

Reply:
It's a recommended vue lint. I guess it's good to have to think about what makes a good key, as the default (loop index) is often not what you actually want.

Reply:
yeah, fair. turns out i had flat config enabled from the dashboard, so lint errors weren't showing up, lol